vtex-apps/render-runtime
# Feature/hydrate on view

First version of partial hydration.

Applies partial hydration on blocks that have their `hydration` value set to `on-view`

Also contains the changes from this PR https://github.com/vtex-apps/render-runtime/pull/489

Required PRs:
https://github.com/vtex-apps/render-runtime/pull/497
https://github.com/vtex/render-server/pull/566
https://github.com/vtex/pages-graphql/pull/348
https://github.com/vtex-apps/store/pull/437
https://github.com/vtex/builder-hub/pull/952

Related PRs:
https://github.com/vtex-apps/store-theme/pull/210
https://github.com/vtex-apps/store-footer/pull/105

### How to test:
https://partialhydration--storecomponents.myvtex.com/
Currently only applied on the home.

Open the page with the network Dev Tools tab filtered by JS.
After the page loads, scroll down slowly, and notice that the assets related to the components that appear into view are being loaded.

Navigate around other pages to check that nothing breaks.

Next, open https://partialhydration--storecomponents.myvtex.com?__disableSSR and verify that it works properly
